dbt/tests/generic_tests/schema.yml

```
@@ -0,0 +1,52 @@
+version: 2
+
+macros:
+  - name: check_column_correlation
+    description: Check the correlation between two columns
+    arguments:
+      - name: column1
+        type: string
+        description: The name of the first column
+      - name: column2
+        type: string
+        description: The name of the second column
+      - name: correlation_coef
+        type: float
+        description: The correlation coefficient to check for
```

Nonblocking:
Is this an exact value or a threshold? & if a threshold, which direction?

Maybe something like 
```suggestion
        description: The minimum correlation coefficient we expect between the two columns.
```